### PR TITLE
MT: Support for Azure China

### PR DESCRIPTION
**Pull request by sskaje**
Before submitting pull request, please ensure that:

- [x] Every commit has a message which describes it
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any code changes come with test case
- [x] Any new functionality is covered by user documentation

### CODE REVIEW

**Comment by nijel** [comment]
Thanks for your contribution! Can you please fix the lint and documentation warnings?

**Comment by sskaje** [comment]
@nijel any suggestion on the Lint / flake8 one?

It was long line but still not passed after wrapping into multiple lines

File: weblate/machinery/microsoft.py [hunk 1]
```diff
@@ -61,7 +59,9 @@ def __init__(self):
             region = "{}.".format(settings.MT_MICROSOFT_REGION)
 
         self._cognitive_token_url = TOKEN_URL.format(
-            region, settings.MT_MICROSOFT_COGNITIVE_KEY
+            region,
+            settings.MT_MICROSOFT_ENDPOINT_URL,
+            settings.MT_MICROSOFT_COGNITIVE_KEY
```
Comment (nijel) [inline weblate/machinery/microsoft.py:64]:
```suggestion
            settings.MT_MICROSOFT_COGNITIVE_KEY,
```

**Comment by sskaje** [comment]
@nijel there's another, hmm, bug, the language mapping in microsoft.py for Chinese should be updated.

Should I wait for this PR merged and then start another PR?

```

    language_map = {
        "zh-hant": "zh-Hant",
        "zh-hans": "zh-Hans",
        "zh-tw": "zh-Hant",
        "zh-cn": "zh-Hans",
        "tlh-qaak": "tlh-Qaak",
        "nb": "no",
        "bs-latn": "bs-Latn",
        "sr-latn": "sr-Latn",
        "sr-cyrl": "sr-Cyrl",
    }

```

I debugged and get the API response from download_languages() 

```
[2020-05-20 08:23:29,106: WARNING/ForkPoolWorker-2] translate: self.supported_languages={'pl', 'mi', 'nl', 'fil', 'yua', 'de', 'zh-Hans', 'it', 'sm', 'ta', 'tlh-Latn', 'hi', 'fi', 'mr', 'el', 'is', 'he', 'ca', 'to', 'hr', 'ht', 'lv', 'yue', 'gu', 'ja', 'pt-pt', 'sk', 'th', 'ty', 'sr-Cyrl', 'zh-Hant', 'pa', 'sl', 'mww', 'uk', 'ar', 'cs', 'ms', 'mg', 'af', 'cy', 'fr', 'es', 'tlh-Piqd', 'ko', 'bs', 'id', 'sv', 'da', 'ur', 'nb', 'otq', 'ml', 'sr-Latn', 'en', 'sw', 'fj', 'fa', 'mt', 'bn', 'pt', 'tr', 'bg', 'et', 'hu', 'ro', 'ga', 'lt', 'vi', 'kn', 'ru', 'te'}

```

And the Azure document: https://docs.microsoft.com/en-us/azure/cognitive-services/translator/language-support


BTW, the bs-latn is not supported now.

**Comment by nijel** [comment]
Please do that in separate PR, I will merge this one now.

It shouldn't matter that it contains languages that are not supported, that is still checked. The language map is used to translate Weblate language codes to the service ones.

**Pull request merged**